tests/dogtag/pytest-ansible/provision/osp_provision.py

```diff
@@ -0,0 +1,192 @@
+def create_connection(auth_url, region, project_name, username, password):
+    return openstack.connect(auth_url=auth_url,
+                             project_name=project_name,
+                             username=username,
+                             password=password,
+                             user_domain_name=constants.DOMAIN_NAME,
+                             project_domain_name=constants.DOMAIN_NAME,
+                             region_name=region,
+                             app_name='RHCS_QE',
+                             app_version='1.0')
+
+
+def spawn_server(conn, image_name, server_type='master'):
+    print("Spawning server: ")
+
+    if server_type == 'master':
+        SERVER_NAME = 'OSP_API_Master_{}'.format(random.randint(999, 99999))
+    else:
+        SERVER_NAME = 'OSP_API_Client_{}'.format(random.randint(999, 99999))
+    image = conn.compute.find_image(image_name)
+    flavor = conn.compute.find_flavor(constants.CI_FLAVOR_NAME)
+    try:
+        server = conn.compute.create_server(name=SERVER_NAME, image_id=image.id,
+                                            flavor_id=flavor.id, networks=[{"uuid": OS_NETWORK_ID}],
+                                            key_name=constants.OS_KEYPAIR_NAME)
+
+        server = conn.compute.wait_for_server(server, status='ACTIVE', failures='ERROR', interval=2, wait=240)
+        return server
+    except Exception as e:
+        print(e)
+        sys.exit(1)
+
+
+def delete_server(conn, server_name):
+    print("Removing Server")
+    server = conn.compute.find_server(server_name)
+    conn.compute.delete_server(server)
+
+
+def list_images(conn, name):
+    print("List Images:")
+    image_list = []
+    for image in conn.image.images():
+        if ("latest" in image.name and image.name.startswith(name)) or (name in image.name):
+            image_list.append(image.name)
+            print("Found Image: {}".format(image.name))
+    return image_list
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+@click.option('--image', default='Fedora 31', help="Image to provision. (Ex. rhel-8.1.0)")
+@click.option('--inventory', default='openstack_inventory', help='Generate inventory file.')
+@click.option('--image-type', type=click.Choice(['nightly', 'production']), default='nightly',
+              help="Image type (nightly, production)")
+@click.option('--server-type', type=click.Choice(['master', 'client']), default='server',
+              help="Machine type (master, client)")
+@click.option('-v', '--verbose', is_flag=True, help="Verbose")
+def up(inventory, image, image_type, server_type, verbose):
+    if verbose:
+        openstack.enable_logging(True, stream=sys.stdout)
+    if 'rhel-7' in image.lower() or 'fedora' in image.lower():
+        python_interpreter = ''
+    else:
+        python_interpreter = 'ansible_python_interpreter=/usr/libexec/platform-python'
+    OS_PASSWORD = os.getenv('OS_PASSWORD')
+    if not image:
+        os_name = 'rhel-8.1.0'
```

Comment:
The `@click.option('--image')` says that the default is Fedora 31.

Reply:
Done changes